testsuite/classlibrary/TestUGen_RTAlloc.sc

```diff
@@ -1,326 +1,325 @@
-		server = Server(this.class.name);
-		server.options.memSize = 1024;
+	*initClass {
+		passVerbosity = UnitTest.brief;
+	}
+
+	*setUpClass {
+		server = Server(this.name);
+		server.options.sampleRate = 48000;
 		server.options.blockSize = 64;
-		this.bootServer(server);
+		server.options.memSize = 2 ** 13; // scsynth default
+		// - tests fail with memSize < 256 (GVerb fails)
+		// - testing with memSize >= 2 ** 20 would require multiple allocations
+		//   (allocation size itself can overflow size_t for some UGens)
+		server.bootSync;
 	}
 
-	tearDown {
+	*tearDownClass {
 		server.quit.remove;
 	}
 
-	test_allocFail_clearUnit {
-		var cond = CondVar(), output = -1;
-		{ DelayN.ar(DC.ar(1), 100); }.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
-			cond.signalOne();
-		});
+	// Async helpers
 
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 0.0, "a failed RTAlloc should clear unit outputs")
-	}
-
-	test_allocFail_continueProcessing_sameNode {
-		var cond = CondVar(), output = -1;
-		{ DelayN.ar(Silent.ar, 100); DC.ar(1)}.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
+	awaitSynthOutput { |synthFn, runtimeSamples=1, timeout=1|
+		var cond = CondVar(), synthOutput = nil;
+		var runtimeSeconds = runtimeSamples / server.sampleRate;
+		// allow extra timeout for longer tests (e.g. testFFT with big memSize)
+		if (runtimeSeconds > timeout) { 
+			timeout = runtimeSeconds + timeout
+		};
+		synthFn.loadToFloatArray(runtimeSeconds, server) { |out|
+			synthOutput = out;
 			cond.signalOne();
-		});
-
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 1.0, "a failed RTAlloc should not block other units in the same node")
+		};
+		cond.waitFor(timeout) { synthOutput.notNil };
+		^synthOutput;
 	}
 
-	test_allocFail_continueProcessing_otherNodes {
-		var cond = CondVar(), output = -1;
-		server.makeBundle(nil) {
-			{ DelayN.ar(Silent.ar, 100) }.play(server);
-			{ DC.ar(1) }.loadToFloatArray(1 / server.sampleRate, server, {|val|
-				output = val[0];
-				cond.signalOne();
-			});
+	assertAllocPass { |name, func, runtimeSamples|
+		var out = this.awaitSynthOutput(func, runtimeSamples);
+		if (out.isNil) {
+			this.assert(false, "% allocPass test should complete".format(name))
+		} {
+			this.assert(out.sum != 0,
+				"% allocPass test should produce non-zero output".format(name)
+			);
 		};
-
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 1.0, "a failed RTAlloc should not block other nodes in the processing chain")
 	}
 
-	test_allocFail_setDoneFlag {
-		var cond = CondVar(), output = -1;
-		{
-			K2A.ar(Done.kr(DelayN.ar(Silent.ar, 100)))
-		}.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
-			cond.signalOne();
-		});
-
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 1.0, "a failed RTAlloc should set the Done flag")
+	assertAllocFail { |name, func, runtimeSamples|
```

Comment:
To be more explicit about timing, this could perhaps be called `awaitAllocFail`.